### PR TITLE
expose stat info on process

### PR DESCRIPTION
**Pull request by ahmetb**

### CODE REVIEW

File: pstree.go [hunk 1]
```diff
@@ -71,60 +99,32 @@ func scan(dir string) (Process, error) {
...
-		priority  int64  // priority
-		nice      int64  // the nice value
-		nthreads  int64  // number of threads in this process
-		itrealval int64  // time in jiffies before next SIGALRM is sent to the process dure to an interval timer
-		starttime int64  // time the process started after system boot in clock ticks
-		vsize     uint64 // virtual memory size in bytes
-		rss       int64  // resident set size: number of pages the process has in real memory
-	}{}
-
+	var stat ProcessStat
```
Comment (sbinet) [inline pstree.go]:
```suggestion
	var proc Process
```
and then fill directly `proc.Stat` below.

File: pstree.go [hunk 2]
```diff
@@ -71,60 +99,32 @@ func scan(dir string) (Process, error) {
...
+		&stat.Nthreads,
+		&stat.Itrealval, &stat.Starttime,
+		&stat.Vsize, &stat.Rss,
 	)
 	if err != nil {
 		return Process{}, err
 	}
 
-	name := stat.comm
+	name := stat.Comm
```
Comment (sbinet) [inline pstree.go]:
```suggestion
	proc.Name = stat.Comm
```
and use that below.

File: pstree.go [hunk 3]
```diff
@@ -71,60 +99,32 @@ func scan(dir string) (Process, error) {
...
 	if strings.HasPrefix(name, "(") && strings.HasSuffix(name, ")") {
 		name = name[1 : len(name)-1]
 	}
 	return Process{
-		Name:   name,
-		Pid:    stat.pid,
-		Parent: stat.ppid,
+		Name: name,
```
Comment (sbinet) [inline pstree.go]: and then just `return proc, err`

**Review by sbinet — COMMENTED** [review]
just a couple of nit-picks.

File: pstree.go (hunk 2 above)
Comment (ahmetb) [reply pstree.go]: I feel like we should entirely eliminate `name` field.
What does removing the parantheses do? Why is this field needed?

Comment (sbinet) [reply pstree.go]: IIRC, this is just used for nice(r) print outs.

Comment (ahmetb) [reply pstree.go]: proc(5) manpage says:
```
              (2) comm  %s
                        The filename of the executable, in parentheses.
                        This is visible whether or not the executable is
                        swapped out.
```

could this actually be something like using parentheses to indicate begin/end of a string as the executable filename may contain spaces?

**Review by sbinet — APPROVED** [review]

**Comment by sbinet** [comment]
thanks a lot.

**Pull request merged**